layers.go

```diff
@@ -823,33 +823,45 @@ func (r *layerStore) Diff(from, to string, options *DiffOptions) (io.ReadCloser,
 	if options != nil && options.Compression != nil {
 		compression = *options.Compression
 	}
+	maybeCompressReadCloser := func(rc io.ReadCloser) (io.ReadCloser, error) {
+		if compression == archive.Uncompressed {
```

Comment:
Depending on whether or not we're asked to compress the diff, this function either returns the passed-in io.ReadCloser, or an io.ReadCloser that returns a compressed version of what the passed-in one would supply to readers.  Adding that in a comment.